### PR TITLE
Bluetooth: Host: Introduce BT_CONN_PARAM_ANY

### PR DESCRIPTION
**Pull request by carlescufi**
Some controllers support additional connection parameter ranges beyond what is described in the specification. Enabling this new option allows the application to set any value to all connection parameters. Tbe Host will perform no limits nor consistency checks on any of the connection parameters (conn interval min and max, latency and timeou). However, the Host will still use numerical comparisons between the min and max connection intervals in order to verify whether the desired parameters have been established in the connection.

### CODE REVIEW

**Comment by jori-nordic** [comment]
@martintv can you check that this works with the llpm sample downstream?

**Review by jori-nordic — DISMISSED** [review]

File: subsys/bluetooth/host/Kconfig.gatt [hunk 1]
```diff
@@ -223,26 +223,26 @@ if BT_GAP_PERIPHERAL_PREF_PARAMS
...
 	int "Peripheral preferred peripheral latency in Connection Intervals"
 	default 0
-	range 0 499
+	range 0 499 if !BT_CONN_PARAM_ANY
 
 config BT_PERIPHERAL_PREF_TIMEOUT
 	int "Peripheral preferred supervision timeout in 10ms units"
 	default 42
-	range 10 65535
+	range 10 65535 if !BT_CONN_PARAM_ANY
```
Comment (Thalley) [inline subsys/bluetooth/host/Kconfig.gatt:245]: Outside the scope of this PR, but the max values of `BT_PERIPHERAL_PREF_MIN_INT`, `BT_PERIPHERAL_PREF_MAX_INT` and `BT_PERIPHERAL_PREF_TIMEOUT` are actually above the values from the HCI spec, and we should consider to set them to more sane values :)

**Review by Thalley — APPROVED** [review]
The fix seems well implemented, and it seems to cover all cases (conn_create, conn_param_update send and conn_param_update recv). 

One thing we could consider is still guarding against interval = 0 or supervision timeout = 0 as neither of those will never make any sense

**Comment by carlescufi** [comment]
> One thing we could consider is still guarding against interval = 0 or supervision timeout = 0 as neither of those will never make any sense

Thanks for the review @Thalley!
It's a good point but I rather not make any assumptions at all :)

Regarding the max values, sure, I will send a follow-up PR after this one.

**Review by jori-nordic — APPROVED** [review]

**Review by jhedberg — APPROVED** [review]

**Review by fabiobaltieri — APPROVED** [review]

**Pull request merged**